### PR TITLE
Move currentSnapLocation to RoutePresenter

### PR DESCRIPTION
**Pull request by sarahlensing**
- Persist currentSnapLocation across screen rotations
- Center map if device rotated in route mode
- Some Kotlin 1.0.1 cleanup

### CODE REVIEW

File: app/src/main/kotlin/com/mapzen/erasermap/view/RouteModeView.kt [hunk 1]
```diff
@@ -431,6 +433,12 @@ class RouteModeView : LinearLayout, RouteViewController, ViewPager.OnPageChangeL
         initInstructionAdapter()
         this.visibility = View.VISIBLE
         routePresenter.onRouteResume(route)
+        Handler().postDelayed( {
+            if (routePresenter.currentSnapLocation != null) {
+                showRouteIcon(routePresenter.currentSnapLocation!!)
```
Comment (ecgreb) [inline app/src/main/kotlin/com/mapzen/erasermap/view/RouteModeView.kt]: Introducing a read-only local variable might be a better way to express this without resorting to forced de-reference using `!!` operator:

```kotlin
        Handler().postDelayed( {
            val currentSnapLocation = routePresenter.currentSnapLocation
            if (currentSnapLocation != null) {
                showRouteIcon(currentSnapLocation)
            }
            centerMapOnCurrentLocation()
        }, 100)
```

File: app/src/main/kotlin/com/mapzen/erasermap/presenter/RoutePresenter.kt [hunk 2]
```diff
@@ -6,26 +6,27 @@ import com.mapzen.erasermap.view.RouteViewController
...
-    public fun onInstructionPagerTouch()
-    public fun onInstructionSelected(instruction: Instruction)
-    public fun onUpdateSnapLocation(location: Location)
-    public fun onRouteClear()
-    public fun onMapListToggleClick(state: MapListToggleButton.MapListState)
-    public fun onRouteCancelButtonClick()
-    public fun mapZoomLevelForCenterMapOnLocation(location: Location): Float
-    public fun isTrackingCurrentLocation(): Boolean
-    public fun onSetCurrentInstruction(index: Int)
+    fun onLocationChanged(location: Location)
```
Comment (ecgreb) [inline app/src/main/kotlin/com/mapzen/erasermap/presenter/RoutePresenter.kt]: Without all the redundant modifiers these are so much more `fun` :)

File: app/src/main/kotlin/com/mapzen/erasermap/view/RouteModeView.kt (hunk 1 above)
Comment (sarahlensing) [inline app/src/main/kotlin/com/mapzen/erasermap/view/RouteModeView.kt]: Good point, let me update that now

Comment (sarahlensing) [inline app/src/main/kotlin/com/mapzen/erasermap/view/RouteModeView.kt]: Fixed

File: app/src/main/kotlin/com/mapzen/erasermap/presenter/RoutePresenter.kt (hunk 2 above)
Comment (sarahlensing) [inline app/src/main/kotlin/com/mapzen/erasermap/presenter/RoutePresenter.kt]: :P

**Pull request merged**